### ISSUE TITLE
Split renewable upstream into separate system processes

### ISSUE THREAD

**Issue by WesIngwersen** [USEPA/ElectricityLCI#150]
Renewable upstream should, like the fossil plant construction upstream, be in separate system processes and not combined along with generation emissions into a single process, so this in consistent, and the sources of these emissions can be separated.

This is the function producing a combined dataset right now

https://github.com/USEPA/ElectricityLCI/blob/2232c41f2cb4fd333ad59c8710aa55906e6a7ed3/electricitylci/__init__.py#L375

**Comment by dt-woods** [comment]
Note that the negative water-to-water emissions are actually inputs in these models, which should account for #147.

**Comment by dt-woods** [comment]
Notable updates:

- https://github.com/USEPA/ElectricityLCI/commit/f77d1a42f6e9c95cf26380fd63434f0525760adb, addition of solar PV, solar thermal, and wind farm construction LCIs
- https://github.com/USEPA/ElectricityLCI/commit/27fb32fd32423ce87056329ef3cd2f08659a0251, solar PV, solar thermal, and wind farm O&M LCIs; update methods for reading in construction and O&M inventories in solar_thermal_upstream.py, solar_upstream.py, and wind_upstream.py
- https://github.com/USEPA/ElectricityLCI/commit/52ca6e421ececbdddc4c1f6a91650a670bbca371, add stage code to generation.py and search for both Electricity and Construction generation processes in generation_mix.py.
- https://github.com/USEPA/ElectricityLCI/commit/897a033a035996611ce8516a97b0c4794b13e21b, ignore indexes when concatenating construction and O&M inventory data frames
- https://github.com/USEPA/ElectricityLCI/commit/0cb2f4a023b0442746cb53ccbde08f8d4dd9f9b0, fix technosphere providers in olcaschema_genprocess method in generation.py

**Comment by dt-woods** [comment]
One small step for mankind:

<img width="1221" alt="Screenshot 2024-10-11 at 21 11 12" src="https://github.com/user-attachments/assets/29a5ec34-94a9-4be4-8fdd-7421b552fc8e">

Key advancements:

- Separated construction and electricity generation processes for solarthermal
- Resources listed in the correct side of the exchange table
- Electricity input as a product flow

Challenges:

- No units associated with Electricity (it should be MWh under 'Energy')

**Comment by dt-woods** [comment]
@m-jamieson, I think I got solarthermal working (see https://github.com/KeyLogicLCA/ElectricityLCI/commit/9b7b34641eb46b4019ffb28e42bcc244b89e50b3). I plan to tackle solar PV and wind farms next.

**Comment by dt-woods** [comment]
Sanity check.

The 2020 renewable electricity generation models (i.e., solar PV, solar thermal, and wind farm) have their upstream (construction) and at plant (operations and maintenance) inventories separated. These are provided in emissions streams in units of kg (with an addition of an Electricity consumption column in MWh). 

I've heard some conflicting info on how plant construction is to be incorporated into the  O&M. 

For coal and natural gas, the inventories were total plant, such that the emissions were divided by the estimated lifetime of the plant to get "per year" emission amounts.

Are the renewables construction flows per year? I think so.

Are the renewables O&M flows per MWh? I think so, because we are scaling them by annual generation.

**Comment by dt-woods** [comment]
**! It appears our construction and O&M inventory CSV files are too big for the repository (45 MB and 24 MB, respectively) !**

**Comment by bl-young** [comment]
Perhaps you could store them on the EPA Data Commons like we do for other data objects (e.g. stewi, fedefl) in an eLCI folder if @WesIngwersen is ok with that (currently he's out of town)

**Comment by dt-woods** [comment]
The problem I'm running into with issues/150 branch is that "Construction - FUEL - BA" processes are popping up for non-renewable fuels (see screenshot).

<img width="640" alt="Screenshot 2024-10-18 at 17 05 09" src="https://github.com/user-attachments/assets/6e6122b4-6d08-43b1-8cbb-f0bcaaeed19b">

A possible reason for this may be that the new renewable inventories have fuel categories that are mis-aligned with EIA.

In get_gen_plus_netl() method found in \_\_init\_\_.py, if you run each line up through `combined_gen`, you'll see a few instances of this:

```py
In: for fid in combined_gen['FacilityID'].unique():
...     tmp_df = combined_gen.query("FacilityID == %d" % fid)
...     f_cat = tmp_df['FuelCategory'].unique()
...     if len(f_cat) > 1:
...         print(fid, len(f_cat), f_cat)
Out:
90 2 ['OIL' 'WIND']
141 2 ['GAS' 'SOLAR']
2953 2 ['GAS' 'SOLAR']
6125 2 ['OIL' 'SOLAR']
8223 2 ['COAL' 'SOLAR']
56322 2 ['WIND' 'SOLAR']
57775 2 ['WIND' 'SOLAR']
58236 2 ['GAS' 'SOLAR']
59825 2 ['WIND' 'SOLAR']
60054 2 ['GAS' 'SOLAR']
60785 2 ['GEOTHERMAL' 'SOLAR']
62442 2 ['WIND' 'SOLAR']
63689 2 ['SOLAR' 'WIND']
10823 2 ['SOLAR' 'WIND']
```

It helps explain the fuel categories where we're seeing this crop up, but it doesn't explain why solar thermal is always the process that gets its construction inventory matched.

**Comment by dt-woods** [comment]
My new plan of action is the following:

Separate construction processes into `get_upstream_process_df` and O&M processes into `get_generation_process_df`. Make the construction fuel type "CONSTRUCTION"; it should get put in the right place (see upstream_dict.py). The question is what is the amount? It should be 1 item, which is equivalent to 1 year of construction, which is the normalization of the construction inventories. After this, remove the double search for construction and electricity in generation_mix.py's olcaschema_genmix. Cross your fingers that the stage code sorts out the product flow for the renewable construction processes.

**Comment by dt-woods** [comment]
^^^ I forgot that we're dealing with facility-level construction inventories, which need to be aggregated to their appropriate region. To avoid over-complicating this, for now, I think I will just make a U.S. average construction inventory for the new renewables. We can discuss how to regionalize construction LCIs in the next iteration (e.g., what's the minimum number of regions necessary to capture the differences, if any?).

**Comment by dt-woods** [comment]
For the U.S. average construction inventories, consider the following approach:

1. Read in the facility-based construction inventories. 
2. Create a scaling coefficient for each facility based on its annual generation divided by the U.S. annual total generation
3. Use the scaling coefficient and multiply it against each facility's emissions to create a scaled inventory.
4. Sum emissions by flow type and category (all emissions are all in kilograms) across all facilities—dropping any NaNs from each array—to make a U.S. average construction inventory.

**Comment by m-jamieson** [comment]
> ^^^ I forgot that we're dealing with facility-level construction inventories, which need to be aggregated to their appropriate region. To avoid over-complicating this, for now, I think I will just make a U.S. average construction inventory for the new renewables. We can discuss how to regionalize construction LCIs in the next iteration (e.g., what's the minimum number of regions necessary to capture the differences, if any?).

If you're following the plan of having the construction inventories be treated like the gas and coal plants, I'm not sure I see why aggregating them is an issue. The gas and coal plants were also at facility-level and get aggregated accordingly.

**Comment by dt-woods** [comment]
Good point and I have amended that in my latest revision, which does the agg'ing to create the following:

<img width="1493" alt="Screenshot 2024-10-22 at 17 54 43" src="https://github.com/user-attachments/assets/44adcc76-bad9-4921-b963-15230d183ad9">

Follow along here: https://github.com/USEPA/ElectricityLCI/discussions/266

**Comment by dt-woods** [comment]
There are two to-do items:

- [ ] Add solar PV and solar thermal construction to power_plant_construction.py
- [ ] Replace solar PV and solar thermal power plant inventories in get_gen_plus_netl w/ their O&M methods

**Comment by dt-woods** [comment]
Now, the only trouble I see is that eLCI's documentation is linking the renewable construction with coal & gas. This occurs in `_process_table_creation_gen` method in upstream_dict.py, which calls `process_description_creation` in process_dictionary_writer.py. This boils down to calling a specific entry in the metadata dictionary defined in "process_metadata.yml" in the data directory, which contains only one entry for "construction_upstream".

**Comment by m-jamieson** [comment]
> Now, the only trouble I see is that eLCI's documentation is linking the renewable construction with coal & gas. This occurs in `_process_table_creation_gen` method in upstream_dict.py, which calls `process_description_creation` in process_dictionary_writer.py. This boils down to calling a specific entry in the metadata dictionary defined in "process_metadata.yml" in the data directory, which contains only one entry for "construction_upstream".

Well, the really easy (lazy) fix would be to just add the renewable construction documentation to make one long statement covering fossil through renewable.

Ideally, we would have separate processes - maybe add as a separate issue, so that we can move ahead with the easy fix now.

**Comment by dt-woods** [comment]
Funny, I started doing that and stopped. We'll need new language to add to
the metadata yaml.
